invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -365,20 +366,86 @@ def view(**kwargs):
     return view
 
 
-def add_signposting(f):
-    """Add signposting link to view's response headers."""
+def _get_header(rel, value, link_type=None):
+    header = f'<{value}> ; rel="{rel}"'
+    if link_type:
+        header += f' ; type="{link_type}"'
+    return header
+
+
+def _get_signposting_collection(pid_value):
+    ui_url = record_url_for(pid_value=pid_value)
+    return _get_header("collection", ui_url, "text/html")
+
+
+def _get_signposting_describes(pid_value):
+    ui_url = record_url_for(pid_value=pid_value)
+    return _get_header("describes", ui_url, "text/html")
+
+
+def _get_signposting_linkset(pid_value):
+    api_url = record_url_for(_app="api", pid_value=pid_value)
+    return _get_header("linkset", api_url, "application/linkset+json")
+
+
+def add_signposting_landing_page(f):
+    """Add signposting links to the landing page view's response headers."""
+
+    @wraps(f)
+    def view(*args, **kwargs):
+        response = make_response(f(*args, **kwargs))
+
+        # Relies on other decorators having operated before it
+        record = kwargs["record"]
+
+        signposting_headers = FAIRSignpostingProfileLvl1Serializer().serialize_object(record.to_dict())
+
+        response.headers["Link"] = signposting_headers
+
+        return response
+
+    return view
+
+
+def add_signposting_content_resources(f):
+    """Add signposting links to the content resources view's response headers."""
 
     @wraps(f)
     def view(*args, **kwargs):
         response = make_response(f(*args, **kwargs))
 
         # Relies on other decorators having operated before it
         pid_value = kwargs["pid_value"]
-        signposting_link = record_url_for(_app="api", pid_value=pid_value)
 
-        response.headers["Link"] = (
-            f'<{signposting_link}> ; rel="linkset" ; type="application/linkset+json"'  # fmt: skip
-        )
+        signposting_headers = [
+            _get_signposting_collection(pid_value),
+            _get_signposting_linkset(pid_value),
+        ]
+
+        response.headers["Link"] = " , ".join(signposting_headers)
+
+        return response
+
+    return view
+
+
+def add_signposting_metadata_resources(f):
+    """Add signposting links to the metadata resources view's response headers."""
+
+    @wraps(f)
+    def view(*args, **kwargs):
+        response = make_response(f(*args, **kwargs))
+
+        # Relies on other decorators having operated before it
+        pid_value = kwargs["pid_value"]
+
+        signposting_headers = [
+            _get_signposting_describes(pid_value),
+            _get_signposting_linkset(pid_value),
+        ]
+
+        response.headers["Link"] = " , ".join(signposting_headers)
+
```

Comment:
Note that, unlike the Landing Page which relies on `invenio_rdm_records.resources.serializers.signposting`, the Content Resources and Metadata Resources are *not* relying on `invenio_rdm_records.resources.serializers.signposting` because:
1. `ContentResourceSchema` and `ContentResourceSchema` expect the record to be passed via `context={"record_dict"}` which makes it more difficult to reuse here.
2. The logic is pretty simple to add only the `collection`, `describes` and `linkset` headers, so re-implementing it here is not that bad.